src/rpc/blockchain.cpp

```diff
@@ -997,7 +997,7 @@ static UniValue gettxoutsetinfo(const JSONRPCRequest& request)
     ::ChainstateActive().ForceFlushStateToDisk();
 
     CCoinsView* coins_view = WITH_LOCK(cs_main, return &ChainstateActive().CoinsDB());
-    if (GetUTXOStats(coins_view, stats, RpcInterruptionPoint)) {
+    if (GetUTXOStats(coins_view, stats, IsRPCRunning() ? RpcInterruptionPoint : [] {})) {
```

Comment:
If the rpc code needs such a workaround for the gui, I'd prefer to at least explain it with a comment

Reply:
Isn't it self-documented: "interrupt RPC only if RPC is running"?

Reply:
Also, it seems racy when the rpc is shut down during flush (which might take a long time)

Maybe the interruption point can be passed in through the `request` similar to how the connman is passed in?

Reply:
> Also, it seems racy when the rpc is shut down during flush (which might take a long time)

Did not get why. `IsRPCRunning()` is checked after `ForceFlushStateToDisk()` call. During GUI shutdown #17659 should ensure the correct `RPCConsole` behavior.

Reply:
* `gettxoutsetinfo`
* `stop` -> Doesn't stop because gettxoutsetinfo is not interruptible
* `gettxoutsetinfo` finishes and server stops

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/19323#issuecomment-646482928).